### ISSUE TITLE
Documentation: Perl example with simple struct does not work

### ISSUE THREAD

**Issue by hakonhagland** [swig/swig#1338]
I am trying out a simple example from the SWIG-3.0 documentation with Perl 5.
This is from section 33.4.5 in the manual:
**example.h**:
```
struct Vector {
  double x, y, z;
};
```
**example.i**:
```
%module example
%{
#include "example.h"
%}
%include "example.h"
```
**Compilation**:
```
swig -perl5 example.i
perl_dir=~/perlbrew/perls/perl-5.26.1/lib/5.26.1/x86_64-linux/CORE 
gcc -fpic -c -Dbool=char -D_GNU_SOURCE -I"$perl_dir" example_wrap.c
gcc -shared example_wrap.o -o example.so
```
**test.pl**:
```
#! /usr/bin/env perl
use strict;
use warnings;
use example;

# This code is copied directly from section 33.4.5 in the SWIG 3.0 manual
my $v = example::new_Vector();
example::Vector_x_set($v, 7.8);          # Change x component
print example::Vector_x_get($v), "\n";   # Get x component
```

The output from running `test.pl` is now:
```
Undefined subroutine &example::new_Vector called at test.pl line 7.
```
However by inspecting the SWIG generated module `example.pm` I can see that the namespace `examplec` is used instead of `example`, so by replacing `example` with `examplec` in the code above, it works fine. Is this a bug in the documentation? Why is the `examplec` namespace introduced?

**My system configuration**:
```
$ uname -a
Linux hakon-Vostro-5568 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic
$ perl --version | head -2
This is perl 5, version 26, subversion 1 (v5.26.1) built for x86_64-linux
$ swig -version
SWIG Version 3.0.12
Compiled with g++ [x86_64-pc-linux-gnu]
Configured options: +pcre
```

**Comment by wsfulton** [comment]
Did you try using the `-noproxy` command line option? @tlby might have some idea what is going wrong.

**Comment by tlby** [comment]
Yes, I guess example has not been updated since proxies were made the default, I verified that with `swig -perl5 -noproxy example.i` works.  With proxies enabled, in `example.pm` you'll find Perl code interacting with the `examplec` functions to create an object oriented interfaces.

The default behavior now would be used like this:
```perl
#! /usr/bin/env perl
use strict;
use warnings;
use example;

my $v = example::Vector->new();
$v->{x} = 7.8;          # Change x component
print $v->{x}, "\n";    # Get x component
```
which people find more natural most of the time.

I should update this documentation.  Thanks for bringing it to my attention!

**Comment by wsfulton** [comment]
@tlby, have you made the documentation changes yet?

**Comment by tlby** [comment]
No, thank you for the ping on this.

**Comment by ojwb** [comment]
@tlby Did you get anywhere with this?  I had a look at updating it but I'm not entirely sure how best to restructure the docs.

**Comment by tlby** [comment]
> I had a look at updating it but I'm not entirely sure how best to restructure the docs.

I empathize, every time I look at narrow updates to the document like this, the need for a large audit/overhaul becomes apparent.  I am a bit overwhelmed right now with other projects right now, but any update is surely better than no updates.  I won't let the perfect be the enemy of the good, and try to iterate here as time permits.
